docs/src/main/asciidoc/datasource.adoc

```diff
@@ -559,44 +549,28 @@ XA transactions on reactive datasources are not supported at the moment.
-through https://www.narayana.io/docs/project/index.html#d5e857[Last Resource Commit Optimization (LRCO)].
-* If you do not need a rollback for one datasource to trigger a rollback for other datasources,
-consider splitting your code into multiple transactions.
-To that end, use xref:transaction.adoc#programmatic-approach[`QuarkusTransaction.requiringNew()`]/xref:transaction.adoc#declarative-approach[`@Transactional(REQUIRES_NEW)`] (preferably)
-or xref:transaction.adoc#legacy-api-approach[`UserTransaction`] (for more complex use cases).
+* Be aware that enabling XA for all datasources _except one_ (and only one) is still supported through https://www.narayana.io/docs/project/index.html#d5e857[Last Resource Commit Optimization (LRCO)].
+* If you do not need a rollback for one datasource to trigger a rollback for other datasources, consider splitting your code into multiple transactions.
+To that end, use xref:transaction.adoc#programmatic-approach[`QuarkusTransaction.requiringNew()`]/xref:transaction.adoc#declarative-approach[`@Transactional(REQUIRES_NEW)`] (preferably) or xref:transaction.adoc#legacy-api-approach[`UserTransaction`] (for more complex use cases).
 
 [CAUTION]
 ====
-As a last resort, and for compatibility with Quarkus 3.8 and earlier,
-you may allow unsafe transaction handling across multiple non-XA datasources
-by setting `quarkus.transaction-manager.unsafe-multiple-last-resources` to `allow`.
-
-With this property set to `allow`, a transaction rollback
-could possibly be applied to only some of the non-XA datasources,
-with other non-XA datasources having already committed their changes,
-leaving your overall system in an inconsistent state.
-
-Alternatively, you can allow the same unsafe behavior,
-but with warnings when it is taken advantage of:
-
-* setting the property to `warn-each`
-would result in logging a warning on *each* offending transaction.
-* setting the property to `warn-first`
-would result in logging a warning on the *first* offending transaction.
-
-We do not recommend using this configuration property,
-and we plan to remove it in the future,
-so you should plan fixing your application accordingly.
-If you think your use case of this feature is valid and this option should be kept around,
-open an issue in the https://github.com/quarkusio/quarkus/issues/new?assignees=&labels=kind%2Fenhancement&projects=&template=feature_request.yml[Quarkus tracker]
+As a last resort, and for compatibility with Quarkus 3.8 and earlier, you might allow unsafe transaction handling across multiple non-XA datasources by setting `quarkus.transaction-manager.unsafe-multiple-last-resources` to `allow`.
+
+With this property set to `allow`, a transaction rollback could possibly be applied to only some of the non-XA datasources, with other non-XA datasources having already committed their changes, leaving your overall system in an inconsistent state.
```

Comment:
The "could be applied" is meant to imply something along the lines of "is likely to happen", or is it meant to talk about something that the user should actively perform? Please use modal with utmost care, if at all.

Reply:
It's something that could potentially happen (but not very often). So I think the wording here is ok.

Reply:
I'd say the rollback could only applied to the last one `non-XA` datasource always.

Reply:
@zhfeng @jmartisk Can we go with this?

With this property set to `allow`, a transaction rollback could only be applied to the last non-XA datasource, while other non-XA datasources could have already committed their changes, potentially leaving your overall system in an inconsistent state.

Reply:
"could" without "possibly" sounds a bit unclear, it sounds like it's in past tense.
I would go with

With this property set to `allow`, it may happen that a transaction rollback will only be applied to the last non-XA datasource, while other non-XA datasources have already committed their changes, potentially leaving your overall system in an inconsistent state.

Reply:
I already stated my POV regarding the usage of `may` in tech docs in the PR.

If I reword it so that I don't have to use "may" + make it more direct and action-oriented + shorter, direct phrasing (that's what the docs should be about if possible), I am getting something like this :) :

Setting this property to `allow` can lead to a transaction rollback which will be applied only to the last non-XA datasource, while other non-XA datasources have already committed their changes, potentially leaving your overall system in an inconsistent state.

Reply:
> Setting this property to allow can lead to a transaction rollback

.. and this sounds to me like you're talking about a one-time event. Like: Once you set the property to `allow`, it can lead to a transaction rollback (one rollback)...

If `may` is the problem (which I personally don't think...) then just replace it with `might`.

Reply:
OK